projects/core/directives/dropdown/dropdown.directive.ts

```diff
@@ -112,5 +120,9 @@ export class TuiDropdownDirective
             this.ref.set(null);
             this.service.remove(ref);
         }
+
+        this.drivers
+            .filter((driver) => driver?.type === this.type)
```

Comment:
No need to check the type as they are all dropdown drivers. However sometimes directives provide themselves as drivers like `TuiDropdownSelection` or `TuiDropdownHover`. Let's put `TODO` on line 59 to think of a better solution later, I suppose it's good enough for now.

Reply:
fixed